packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -1407,10 +1407,42 @@ class ResizeImage extends ImageProvider<ResizeImageKey> {
   }
 }
 
+/// The strategy for [Image.network] and [NetworkImage] to decide whether to
+/// display images in <img> elements in a platform view instead of fetching
+/// bytes.
+///
+/// See [Image.network] for more explanation on the impact.
+///
+/// This option is only effective on the Web platform. Other platforms always
+/// display network images by fetching bytes.
+enum WebImgElementStrategy {
```

Comment:
`img` is a bit of a implementation detail that could theoretically change. For example, we could switch to `<picture>`, SVG, or tainted 2D canvas for rendering CORS images. I'd therefore avoid putting "Img" in the name of the enum.

How about we call it just `SameOriginStrategy`?